x-pack/metricbeat/module/oracle/performance/data.go

```diff
@@ -59,14 +61,23 @@ func (m *MetricSet) transform(in *extractedData) []mb.Event {
 	events := make([]mb.Event, 0)
 
 	for _, v := range bufferCache {
-		events = append(events, mb.Event{MetricSetFields: v})
+		events = append(events, mb.Event{MetricSetFields: v, Host: ServiceNameExtractor(m.BaseMetricSet.HostData().URI)})
 	}
 
-	events = append(events, mb.Event{MetricSetFields: cursorEvent})
+	events = append(events, mb.Event{MetricSetFields: cursorEvent, Host: ServiceNameExtractor(m.BaseMetricSet.HostData().URI)})
 
 	for _, v := range cursorByUsernameAndMachineEvents {
-		events = append(events, mb.Event{MetricSetFields: v})
+		events = append(events, mb.Event{MetricSetFields: v, Host: ServiceNameExtractor(m.BaseMetricSet.HostData().URI)})
 	}
 
 	return events
 }
+
+// ServiceName extracts ip address from host.
+func ServiceNameExtractor(host string) string {
```

Comment:
This name extractor wouldn't work with normal urls, that seem to be also supported by `godror.ParseDSN`. Ideally you should be using the host included in the result of `godror.ParseDSN`.

Reply:
Yes, we tried the same before and it definitely sounds like a better way. But, the problem with it was that it collected the same value for the field `service.address` when we tried passing multiple hosts in the `oracle.yml` configuration.

When we were digging deeper into it after your comment, we were wondering how the Oracle module iterates through multiple hosts provided in the configuration. What is the reason behind passing just `Hosts[0]` [here](https://github.com/elastic/beats/blob/main/x-pack/metricbeat/module/oracle/connection.go#L40)?

We also tried collecting data from two different Oracle hosts, but most of the metrics looked similar to us.

Was the above problem because of this or are we missing something here? Can you please point us in the right direction?

Reply:
> Was the above problem because of this or are we missing something here? Can you please point us in the right direction?

I think this module needs to be refactored to properly use host parsers instead of doing its own parsing of `hosts`. This could be done in a separated PR.

Reply:
Thanks a lot for the detailed explanation, @jsoriano! That resolved the doubts and made things pretty clear.
I have raised an issue for the same, and added it to the existing [Oracle Module Enhancements PRD](https://github.com/elastic/beats/issues/30810).

Issue Link: https://github.com/elastic/beats/issues/31611
I have added details to the issue based on our conversations. Please feel free to update the issue with more details if required or in case I missed anything.

Reply:
> I think this module needs to be refactored to properly use host parsers instead of doing its own parsing of hosts. This could be done in a separated PR.

Raised the PR https://github.com/elastic/beats/pull/31692 to refactor the module to do the same, please take a look at your convenience.